pkg/controller/certificates/trigger/policies/policies.go

```diff
@@ -189,27 +190,29 @@ func CurrentCertificateNearingExpiry(c clock.Clock) Func {
 			return "", "", false
 		}
 
-		return "Renewing", fmt.Sprintf("Renewing certificate as renewal was scheduled at %s", input.Certificate.Status.RenewalTime), true
+		return Renewing, fmt.Sprintf("Renewing certificate as renewal was scheduled at %s", input.Certificate.Status.RenewalTime), true
 	}
 }
 
 // CurrentCertificateHasExpired is used exclusively to check if the current
 // issued certificate has actually expired rather than just nearing expiry.
-func CurrentCertificateHasExpired(input Input) (string, string, bool) {
-	certData := input.Secret.Data[corev1.TLSCertKey]
-	// TODO: replace this with a generic decoder that can handle different
-	//  formats such as JKS, P12 etc (i.e. add proper support for keystores)
-	cert, err := pki.DecodeX509CertificateBytes(certData)
-	if err != nil {
-		// This case should never happen as it should always be caught by the
-		// secretPublicKeysMatch function beforehand, but handle it just in case.
-		return "InvalidCertificate", fmt.Sprintf("Failed to decode stored certificate: %v", err), true
-	}
+func CurrentCertificateHasExpired(c clock.Clock) Func {
+	return func(input Input) (string, string, bool) {
+		certData := input.Secret.Data[corev1.TLSCertKey]
```

Comment:
We should check the bool 'ok' var here too, and exit early if it is false.

Reply:
I was undecisive about this as theoretically that should have been caught by other policy functions earlier in the chain. But yes, safer to check.